src/reporters/build_kite_cli.ts

```diff
@@ -0,0 +1,187 @@
+} from '../common_types';
+import { Journey, Step } from '../dsl';
+import { gatherScreenshots } from './json';
+import Runner from '../core/runner';
+
+export async function writeScreenshotsToPath(
+  screenshotOptions: ScreenshotOptions,
+  outputDir: string,
+  status: StepEndResult['status']
+) {
+  const FAILED_STEPS_PATH = join(outputDir, 'failed_steps');
+  await fs.promises.mkdir(FAILED_STEPS_PATH, { recursive: true });
+
+  await gatherScreenshots(Runner.screenshotPath, async screenshot => {
+    const { data, step } = screenshot;
+    try {
+      const filename = sanitizeFilename(`${step.name}.jpg`);
+
+      if (screenshotOptions === 'on') {
+        fs.writeFileSync(join(outputDir, filename), data, {
+          encoding: 'base64',
+        });
+      } else if (
+        screenshotOptions === 'only-on-failure' &&
+        status === 'failed'
+      ) {
+        fs.writeFileSync(join(FAILED_STEPS_PATH, filename), data, {
+          encoding: 'base64',
+        });
+      }
+    } catch (e) {
+      // eslint-disable-next-line no-console
+      console.log(e);
+    }
+  });
+}
+
+export default class BuildKiteCLIReporter extends BaseReporter {
+  screenshotsPath: string;
+
+  journeys: Map<string, Array<StepEndResult & { name: string }>> = new Map();
+
+  constructor(options: ReporterOptions = {}) {
+    super(options);
+    this.recordVideo = options.recordVideo;
+    this.screenshotsPath = join(CACHE_PATH, 'screenshots');
+  }
+
+  override onJourneyStart(journey: Journey, {}: JourneyStartResult) {
+    this.write(`\n--- Journey: ${journey.name}`);
+  }
+
+  override onStepEnd(journey: Journey, step: Step, result: StepEndResult) {
+    const { status, end, start, error } = result;
+    const message = `${symbols[status]}  Step: '${
+      step.name
+    }' ${status} (${renderDuration((end - start) * 1000)} ms)`;
+    this.write(indent(message));
+    if (error) {
+      this.write(renderError(error));
+    }
+    this.metrics[status]++;
+    if (!this.journeys.has(journey.name)) {
+      this.journeys.set(journey.name, []);
+    }
+    this.journeys.get(journey.name)?.push({ name: step.name, ...result });
+  }
+
+  override async onJourneyEnd(
+    journey: Journey,
+    { error, start, end, status, options }: JourneyEndResult
+  ) {
+    const { screenshots, outputDir } = options;
+    const { failed, succeeded, skipped } = this.metrics;
+    const total = failed + succeeded + skipped;
+    if (total === 0 && error) {
+      this.write(renderError(error));
+    }
+    const message = `${symbols[status]} Took  (${renderDuration(
+      end - start
+    )} seconds)`;
+    this.write(message);
+
+    if (outputDir) {
+      const FAILED_STEPS_PATH = join(outputDir, 'failed_steps');
+      await fs.promises.mkdir(FAILED_STEPS_PATH, { recursive: true });
+
+      await writeScreenshotsToPath(screenshots, outputDir, status);
+    }
+  }
+
+  override onEnd() {
+    const failedJourneys = Array.from(this.journeys.entries()).filter(
```

Comment:
Not sure if this is what you want, I see Step errors getting printed multiple times. Single journey with a step that fails. 
<img width="825" alt="Screenshot 2023-09-01 at 4 33 12 PM" src="https://github.com/elastic/synthetics/assets/3902525/cb1fc5a1-a48e-4656-bb07-ab85c3f7679c">

Reply:
so the idea here is to print all the failed journeys at the end. 
if there are too many journeys, that's usually helpful.